src/Groups/schur_index.jl

```diff
@@ -0,0 +1,520 @@
+#
+# deal with extensions of p-adic fields by elements of abelian number fields
+#
+# For `vals` with conductor N = p^k m, with m coprime to p,
+# we describe Q_p(vals) by (N, stab, [Q_p(N):Q_p]),
+# where stab is the set of prime residues u modulo N
+# such that *u is in Gal(Q_p(N)/Q_p) and chi is fixed by *u
+#
+# We use that Gal(Q_p(N)/Q_p) is the direct product of the group generated
+# by \sigma_p: \zeta_m \mapsto \zeta_m^p, \zeta_{p^k} \mapsto \zeta_{p^k}
+# and the group of maps
+# \sigma_u: \zeta_m \mapsto \zeta_m, \zeta_{p^k} \mapsto \zeta_{p^k}^u,
+# where u is a prime residue modulo p^k.
+#
+# This means that [Q_p(vals):Q_p] == [Q_p(N):Q_p] / length(stab)
+#
+const _Q_p_chi_cache = Dict{Tuple{Int, Vector{Int}, Int},
+                            Tuple{Int, Vector{Int}, Int}}()
+
+function _Q_p_chi(vals::GapObj, p::ZZRingElem)
+  if length(vals) == 0
+    F = GAP.Globals.Rationals
+  else
+    F = GAP.Globals.Field(GAP.Globals.Rationals, vals)
+  end
+  N = GAP.Globals.Conductor(F)
+  if N == 1
+    stab = [1]
+  else
+    stab = Vector{Int}(GAP.Globals.GaloisStabilizer(F))
+  end
+
+  return get!(_Q_p_chi_cache, (N, stab, p)) do
+    N == 1 && return (N, stab, 1)
+
+    ppart, m = ppio(ZZRingElem(N), p)
+    _, a, b = gcdx(ppart, m)
+    r = m == 1 ? 1 : modord(p, m)
+#T `modord(N, 1)` could return 1 (smallest pos. int. i s.t. 1 divides (N^i-1))
+    stab = Int[]
+    sigma_p = mod(1 + a*(p-1)*ppart, N)
+    res = ppart == 1 ? [1] : Vector{Int}(GAP.Globals.PrimeResidues(GapObj(ppart)))
+    for u in res
+      for i in 0:(r-1)
+        sigma = mod(sigma_p^i * (1 + (u-1)*m*b), N)
+        if GAP.Globals.GaloisCyc(vals, GapObj(sigma)) == vals
+          push!(stab, sigma)
+        end
+      end
+    end
+
+    return (N, stab, r*euler_phi(ppart))
+  end
+end
+
+
+# `elm` is an element of a cyclotomic field in GAP.
+# `info` is a description `(N, stab, index)` of the field
+# $F \subseteq \Q_p(N)$ such that all elements in $F$ are fixed under the
+# Galois automorphisms $*k$, for $k \in stab$.
+# (We have $[\Q_p(N):F] = index$, but this is nnot used here.)
+function _membership_test(elm::GAP.Obj, info::Tuple{Int, Vector{Int}, Int})
+  return mod(info[1], GAPWrap.Conductor(elm)) == 0 &&
+         all(k -> GAP.Globals.GaloisCyc(elm, k) == elm, info[2])
+end
+
+
+#############################################################################
+
+"""
+    local_schur_indices(chi::GAPGroupClassFunction; cyclic_defect::Vector{Int} = Int[])
+
+Return the array of pairs `p => m_p(chi)` such that `m_p(chi) > 1` holds.
+The Schur index over the real field is stored at `p = -1` if applicable.
+
+`cyclic_defect` contains primes `p` such that `chi` is known to belong to
+a `p`-block of cyclic defect.
+
+# Examples
+```jldoctest
+julia> G, chi = Oscar.yamada_example(5, 2);
+
+julia> order(G)
+40
+
+julia> local_schur_indices(chi)
+2-element Vector{Pair{Int64, Int64}}:
+ -1 => 2
```

Comment:
this too needs thought: for G-Modules, the local indices can be labled by prime numbers as all conjugate prime ideals behave the same (a property of the Schur-Subgroupof the Brauer group). In general there need to be prime ideals or places here if we want to extend/ use the same interface for A-modules

Reply:
I think the unqualified `local_schur_indices(chi)` for complex characters need not be in sync with the modules over fin.-dim. algebras. The interface here is the same as in Magma, except that Magma uses 0 instead of -1 for the archimedean place. I am leaning towards `0`, since we use this convention already for things like `hilbert_symbol`.

Reply:
Concerning the 0 vs. -1:
I have tried to use the same format as in `experimental/GModule/src/GModule.jl`. I have no preference, it should just be consistent.

Reply:
@fieker are you opposed to changing 0 to -1?